File: server/internal/models/device.go
package models

import (
	"github.com/google/uuid"
	"github.com/juevigrace/diva-server/internal/models/responses"
	"github.com/juevigrace/diva-server/storage"
)

type Device struct {
	ID        uuid.UUID
	Name      string
	CreatedAt int64
	UpdatedAt int64
}

func (d *Device) Response() *responses.DeviceResponse {
	return &responses.DeviceResponse{
		ID:        d.ID.String(),
		Name:      d.Name,
		CreatedAt: d.CreatedAt,
		UpdatedAt: d.UpdatedAt,
	}
}

func DeviceFromDB(row *storage.DivaDevice) *Device {
	return &Device{
		ID:        row.ID,
		Name:      row.Name,
		CreatedAt: row.CreatedAt,
		UpdatedAt: row.UpdatedAt,
	}
}

File: kmp/app/core/models/models-core/src/commonMain/kotlin/com/diva/models/device/Device.kt
package com.diva.models.device

import com.diva.models.api.device.response.DeviceResponse
import kotlin.time.Instant
import kotlin.uuid.ExperimentalUuidApi
import kotlin.uuid.Uuid

@OptIn(ExperimentalUuidApi::class)
data class Device(
    val id: Uuid,
    val name: String,
    val createdAt: Instant,
    val updatedAt: Instant,
) {
    companion object {
        fun fromResponse(response: DeviceResponse): Device {
            return Device(
                id = Uuid.parse(response.id),
                name = response.name,
                createdAt = Instant.fromEpochMilliseconds(response.createdAt),
                updatedAt = Instant.fromEpochMilliseconds(response.updatedAt),
            )
        }
    }
}
